defender/emitVotes.js

```diff
@@ -0,0 +1,241 @@
+              );
+
+    const governor = new ethers.Contract(
+        governorAddress,
+        governorABI,
+        moonbeamProvider,
+    );
+
+    const liveProposals = await governor.liveProposals();
+
+    const client = new Defender(event);
+    const provider = client.relaySigner.getProvider();
+    const signer = await client.relaySigner.getSigner(provider, {
+        speed: 'fast',
+    });
+
+    if (liveProposals.length == 0) {
+        console.log('No live proposals found');
+        return {};
+    }
+
+    const voteCollection = new ethers.Contract(
+        voteCollectionAddress,
+        voteCollectionABI,
+        signer,
+    );
+
+    for (const proposalId of liveProposals) {
+        const state = await governor.state(proposalId);
+        console.log(`Proposal ${proposalId} state: ${state}`);
+
+        const kvStore = new KeyValueStoreClient(event);
+        let kvStoreValue = await kvStore.get(`${network}`);
+        const ids = kvStoreValue?.split(',') || [];
+
+        if (ids.includes(proposalId.toString())) {
+            console.log(`Votes already emitted for proposal ${proposalId}`);
+            continue;
+        }
+
+        if (state == 1) {
+            console.log(
+                `Proposal ${proposalId} is in the cross chain vote collection state`,
+            );
+
+            const proposalInformation =
+                await voteCollection.proposalInformation(proposalId);
+
+            const {notificationClient} = context;
+            let subject;
+            let message;
+
+            // only emit votes if proposal total votes is greater than 0
+            if (proposalInformation[5] > 0) {
+                try {
+                    const bridgeCost = await voteCollection.bridgeCostAll();
+
+                    const gasEstimate =
+                        await voteCollection.estimateGas.emitVotes(proposalId, {
+                            value: bridgeCost,
+                        });
+
+                    // Add a buffer to the gas estimate
+                    const gasLimit = gasEstimate.add(gasEstimate.div(5)); // Adding 20% extra gas as a buffer
+                    await voteCollection.emitVotes(proposalId, {
+                        value: bridgeCost,
+                        gasLimit: gasLimit,
+                    });
+
+                    subject = `Votes emitted for proposal ${proposalId} on ${network}`;
+                    message = `If the proposal reaches quorum, it will be executed automatically once the cross-chain vote collection period finishes.`;
+                } catch (error) {
+                    console.error(
+                        `Error emitting votes for proposal ${proposalId}: ${error.message}`,
+                    );
+                    subject = `Error emitting votes for proposal ${proposalId} on ${network}`;
+                    message = `Error: ${error.message}. Manual intervention required.`;
+                }
+
+                if (kvStoreValue) {
```

Comment:
should we store the proposal id in key value store if the `emitVotes` function call has failed.